drivers/i2c/i2c_esp32.c

```diff
@@ -671,9 +669,6 @@ static int i2c_esp32_init(const struct device *dev)
 	const struct i2c_esp32_config *config = dev->config;
 	struct i2c_esp32_data *data = dev->data;
 	uint32_t bitrate_cfg = i2c_map_dt_bitrate(config->bitrate);
-	data->clock_dev = device_get_binding(config->clock_name);
```

Comment:
There should be an ASSERT that `config->clock_dev` is ready, either here or in the functions that use it.